doc/source/analysis.rst

```diff
@@ -1,5 +1,400 @@
+- :meth:`Graph.is_dag`
+- :meth:`Graph.is_directed`
+- :meth:`Graph.is_named`
+- :meth:`Graph.is_simple`
+- :meth:`Graph.is_weighted`
+- :meth:`Graph.has_multiple`
+
+Vertex properties
++++++++++++++++++++
+A spectrum of vertex-level properties can be computed. Similarity measures include:
+
+- :meth:`Graph.similarity_dice`
+- :meth:`Graph.similarity_jaccard`
+- :meth:`Graph.similarity_inverse_log_weighted`
+- :meth:`Graph.diversity`
+
+Structural:
+
+- :meth:`Graph.authority_score`
+- :meth:`Graph.betweenness`
+- :meth:`Graph.bibcoupling`
+- :meth:`Graph.closeness`
+- :meth:`Graph.constraint`
+- :meth:`Graph.cocitation`
+- :meth:`Graph.eccentricity`
+- :meth:`Graph.eigenvector_centrality` aka :meth:`Graph.evcent`
+- :meth:`Graph.pagerank`
+- :meth:`Graph.personalized_pagerank`
+- :meth:`Graph.strength`
+
+Connectedness:
+
+- :meth:`Graph.subcomponent`
+- :meth:`Graph.is_separator`
+- :meth:`Graph.is_minimal_separator`
+- :meth:`Graph.biconnected_components`
+- :meth:`Graph.blocks`
+
+Edge properties
++++++++++++++++
+As for vertices, edge properties are implemented. Basic properties include:
+
+- :meth:`Graph.is_loop`
+- :meth:`Graph.is_multiple`
+- :meth:`Graph.is_mutual`
+- :meth:`Graph.count_multiple`
+
+and more complex ones:
+
+- :meth:`Graph.edge_betweenness`
+
+Matrix representations
++++++++++++++++++++++++
+Matrix-related functionality includes:
+
+- :meth:`Graph.get_adjacency`
+- :meth:`Graph.get_adjacency_sparse` (sparase CSR matrix version)
+- :meth:`Graph.laplacian`
+
+Clustering
+++++++++++
+|igraph| includes several approaches to unsupervised graph clustering and community detection:
+
+- :meth:`Graph.clusters` (aka :meth:`Graph.components`): the connected components
+- :meth:`Graph.cohesive_blocks`
+- :meth:`Graph.community_edge_betweenness`
+- :meth:`Graph.community_fastgreedy`
+- :meth:`Graph.community_infomap`
+- :meth:`Graph.community_label_propagation`
+- :meth:`Graph.community_leading_eigenvector`
+- :meth:`Graph.community_leading_eigenvector_naive`
+- :meth:`Graph.community_leiden`
+- :meth:`Graph.community_multilevel` (a version of Louvain)
+- :meth:`Graph.community_optimal_modularity` (exact solution, < 100 vertices)
+- :meth:`Graph.community_spinglass`
+- :meth:`Graph.community_walktrap`
+
+Simplification, permutations and rewiring
++++++++++++++++++++++++++++++++++++++++++
+To simplify a graph (remove multiedges and loops), use :meth:`Graph.simplify`:
+
+>>> g_simple = g.simplify()
+
+To return a directed/undirected copy of the graph, use :meth:`Graph.as_directed` and :meth:`Graph.as_undirected`, respectively.
+
+To permute the order of vertices, you can use :meth:`Graph.permute_vertices`:
+
+>>> g = ig.Tree(6, 2)
+>>> g_perm = g.permute_vertices([1, 0, 2, 3, 4, 5])
+
+The canonical permutation can be obtained via :meth:`Graph.canonical_permutation`, which can then be directly passed to the function above.
+
+To rewire the graph at random while keeping some structural properties, there are:
```

Comment:
Mention what properties these two rewiring methods keep invariant (i.e. edge count and degree sequence)?

Reply:
Improved this a little. They don't seem to keep property fixed exactly, just at the distribution level.